backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CStubsManager.kt

```diff
@@ -0,0 +1,81 @@
+package org.jetbrains.kotlin.backend.konan
+
+import org.jetbrains.kotlin.cli.common.messages.CompilerMessageLocation
+import org.jetbrains.kotlin.cli.common.messages.CompilerMessageSeverity
+import org.jetbrains.kotlin.cli.common.messages.MessageCollector
+import org.jetbrains.kotlin.konan.exec.Command
+import org.jetbrains.kotlin.konan.file.*
+import org.jetbrains.kotlin.konan.target.ClangArgs
+
+class CStubsManager {
+
+    fun getUniqueName(prefix: String) = "$prefix${counter++}"
+
+    fun addStub(kotlinLocation: CompilerMessageLocation?, lines: List<String>) {
+        stubs += Stub(kotlinLocation, lines)
+    }
+
+    fun compile(clang: ClangArgs, messageCollector: MessageCollector, verbose: Boolean): File? {
+        if (stubs.isEmpty()) return null
+
+        val cSource = createTempFile("cstubs", ".c").deleteOnExit()
+        cSource.writeLines(stubs.flatMap { it.lines })
```

Comment:
Shalln't temp files be closed, so that compiler surely see all of them?

Reply:
They definitely must. That's why `writeLines` opens the file, writes the lines and then closes the file.